integration-tests/geocoder/src/test/java/org/apache/camel/quarkus/component/geocoder/it/GeocoderGoogleTest.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.quarkus.component.geocoder.it;
+
+import io.quarkus.test.common.http.TestHTTPEndpoint;
+import io.quarkus.test.junit.QuarkusTest;
+import io.restassured.RestAssured;
+import org.eclipse.microprofile.config.ConfigProvider;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import static org.apache.camel.quarkus.component.geocoder.it.GeocoderGoogleResource.GOOGLE_GEOCODER_API_KEY;
+import static org.hamcrest.Matchers.hasKey;
+
+@QuarkusTest
+@TestHTTPEndpoint(GeocoderGoogleResource.class)
+class GeocoderGoogleTest {
+
+    private static boolean enabled;
+
+    @BeforeAll
+    public static void setup() {
+        // enables the integration tests if an API key exists
+        enabled = ConfigProvider.getConfig()
+                .getOptionalValue(GOOGLE_GEOCODER_API_KEY, String.class).isPresent();
```

Comment:
You can use `@EnabledIfEnvironmentVariable` for disabling the test like we do in https://github.com/apache/camel-quarkus/blob/1175fdc6dd5e6ee4a7782afa56b404803ad2e73d/integration-tests/braintree/src/test/java/org/apache/camel/quarkus/component/braintree/it/BraintreeIT.java#L25

If you do that that way, the test app is not started at all and you actually do not need to check in the app whether property is set. 

I think in case the GOOGLE_GEOCODER_API_KEY is not provided, the test should run against a mock API, something like we do with Telegram and elsewhere. If a mock testcontainer is available we should use it. No need to do this now, I'll file a followup.

Reply:
OK